currencies/rates.go

```diff
@@ -0,0 +1,24 @@
+package currencies
+
+import (
+	"time"
+)
+
+// Rates holds data as represented on http://currency.prebid.org/latest.json
+// note that `DataAsOfRaw` field is needed when parsing remote JSON as the date format if not standard and requires
+// custom parsing to be properly set as Golang time.Time
+type Rates struct {
+	DataAsOfRaw string                        `json:"dataAsOf"`
+	DataAsOf    *time.Time                    `json:"-"`
+	Conversions map[string]map[string]float32 `json:"conversions"`
+}
+
+// ComputeDataAsOf computes `DataAsOf` by parsing `DataAsOfRaw` which is a raw string representing a date
+func (c *Rates) ComputeDataAsOf() error {
```

Comment:
+1 to computing the Time so people don't have to think about the string.
-1 to making callers run `ComputeDataAsOf()` every time after they Unmarshal a response.

You could just implement the [Unmarshaler](https://golang.org/pkg/encoding/json/#Unmarshaler) and do this work inside there. As a bonus it'd also let you remove `DataAsOfRaw`, which doesn't really have much value once it's been parsed.

Reply:
Totally agree, I just forgot to get back to this implementing the `UnMarshaller`.

Reply:
Done